src/Middlewares/SecurityInputFieldMiddleware.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+declare(strict_types=1);
+
+namespace TheCodingMachine\GraphQLite\Middlewares;
+
+use TheCodingMachine\GraphQLite\InputField;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\ExpressionLanguage\ExpressionLanguage;
+use TheCodingMachine\GraphQLite\Annotations\Security;
+use TheCodingMachine\GraphQLite\Parameters\ParameterInterface;
+use TheCodingMachine\GraphQLite\InputFieldDescriptor;
+use TheCodingMachine\GraphQLite\Security\AuthenticationServiceInterface;
+use TheCodingMachine\GraphQLite\Security\AuthorizationServiceInterface;
+use Throwable;
+use Webmozart\Assert\Assert;
+
+use function array_combine;
+use function array_keys;
+
+/**
+ * A field input middleware that reads "Security" Symfony annotations.
+ * it is the equivalent to the SecurityFieldMiddleware.
+ */
+class SecurityInputFieldMiddleware implements InputFieldMiddlewareInterface
+{
+    /** @var ExpressionLanguage */
+    private $language;
+    /** @var AuthenticationServiceInterface */
+    private $authenticationService;
+    /** @var LoggerInterface|null */
+    /*private $logger;*/
+    /** @var AuthorizationServiceInterface */
+    private $authorizationService;
+
+    public function __construct(ExpressionLanguage $language, AuthenticationServiceInterface $authenticationService, AuthorizationServiceInterface $authorizationService/*, ?LoggerInterface $logger = null*/)
+    {
+        $this->language = $language;
+        $this->authenticationService = $authenticationService;
+        $this->authorizationService = $authorizationService;
+        /*$this->logger = $logger;*/
```

Comment:
Unused code

Reply:
connected to the logger stuff above

Reply:
Let's add a comment for the reasoning for it being there.